Admin/AbstractAdmin.php

```diff
@@ -1893,9 +1905,19 @@ public function getFilterFieldDescriptions()
      */
     public function addChild(AdminInterface $child)
     {
+        for ($parentAdmin = $this; null !== $parentAdmin; $parentAdmin = $parentAdmin->getParent()) {
+            if ($parentAdmin->getCode() !== $child->getCode()) {
+                continue;
+            }
+
+            throw new \RuntimeException(sprintf(
+                'Circular reference detected! The child admin `%s` is already in the parent tree of the `%s` admin.',
+                $child->getCode(), $this->getCode()
+            ));
+        }
+
```

Comment:
I'm aware that I am not checking for circular references on calls of `setParent(...)` but this would require to iterates over all children. Is it really worth it?